### PR TITLE
deleting worker from workers list on deleting a user

### PR DESCRIPTION
**Pull request by mahakporwal02**
## Rationale

[//]: # (Briefly explain the reason behind this change.)
The worker does not delete from workers list even when its corresponding user is deleted

#556 
## Changes
Added code to delete worker when user is being deleted

### CODE REVIEW

File: dispatcher/backend/src/routes/users/user.py [hunk 1]
```diff
@@ -129,6 +130,8 @@ def patch(self, token: AccessToken.Payload, username: str):
     @url_object_id("username")
     def delete(self, token: AccessToken.Payload, username: str):
         # delete user
+        if get_role_for(Users().find_one({"username":username}).get("scope")) == "worker":
```
Comment (rgaudin) [inline dispatcher/backend/src/routes/users/user.py:133]: Inserting new code in between a comment and the code that's related to it is misleading.

File: dispatcher/backend/src/routes/users/user.py [hunk 2]
```diff
@@ -6,6 +6,7 @@
 from marshmallow import ValidationError
 
 from common.mongo import Users
+from common.mongo import Workers
```
Comment (rgaudin) [inline dispatcher/backend/src/routes/users/user.py:9]: You should share that import with the previous one

File: dispatcher/backend/src/routes/users/user.py (hunk 1 above)
Comment (rgaudin) [reply dispatcher/backend/src/routes/users/user.py:133]: You should use two steps so that the workers delete is only happening after the user is deleted

Comment (rgaudin) [reply dispatcher/backend/src/routes/users/user.py:133]: This will crash if there is no User with that username as `find_one` will return `None`

Comment (rgaudin) [reply dispatcher/backend/src/routes/users/user.py:133]: You didn't respect the black formatting (`pip install black` then `black <yourfile`)

**Review by rgaudin — CHANGES_REQUESTED** [review]
Thanks ; strategy's good but there are a number of stuff to fix

File: dispatcher/backend/src/routes/users/user.py (hunk 2 above)
Comment (mahakporwal02) [reply dispatcher/backend/src/routes/users/user.py]: @rgaudin I have done the specified changes

**Review by rgaudin — APPROVED** [review]

**Pull request merged**